lazy-pages/src/sys/unix.rs

```diff
@@ -67,9 +68,21 @@ extern "C" fn handle_sigsegv(sig: i32, info: *mut siginfo_t, ucontext: *mut c_vo
 
 pub unsafe fn setup_signal_handler() -> io::Result<()> {
     let handler = signal::SigHandler::SigAction(handle_sigsegv);
+
+    // Set additional SA_ONSTACK and SA_NODEFER to avoid problems with wasmer executor.
+    // See comment from shorturl.at/KMO68 :
+    // ```
+    //  SA_ONSTACK allows us to handle signals on an alternate stack,
```

Comment:
It won't work without [sigaltstack(2)](https://man7.org/linux/man-pages/man2/sigaltstack.2.html). After the change it will also fix #1444

Reply:
> It won't work without sigaltstack(2)
Are you sure about that ? As you can see `runtime-spec` and `pallet` test works well. They don't use `wasmer` as executor. Also I just check it for `wasmi` host executor - all goes right as well. I think it works on a common stack without sigaltstack 

> After the change it will also fix #1444
Really ? Cannot understand why))

Reply:
> test works well

Hm, maybe because:

> Rust installs an alternate stack with sigaltstack, so we rely on that.

> > After the change it will also fix https://github.com/gear-tech/gear/issues/1444
>
> Really ? Cannot understand why))

If we will install a new stack for each thread - signal won't overlap anymore

Reply:
Ok, I see now